xla/stream_executor/rocm/BUILD

```diff
@@ -306,7 +307,7 @@ cc_library(
         "//xla/stream_executor:event",
         "//xla/stream_executor:plugin_registry",
         "//xla/stream_executor:scratch_allocator",
-        "//xla/stream_executor:stream_executor_pimpl_header",
```

Comment:
This likely has to be `//xla/stream_executor` or `//xla/stream_executor:stream_executor_impl`, I'll be removing `pimpl` targets shortly